### ISSUE TITLE
Undefined variable: _COOKIE 

### ISSUE THREAD

**Issue by molchanov1983** [yiisoft/yii#4138]
Hello
I have this error sometimes and i don't know how to reproduce and when it happens
Stack: apache 2.4 php 7.1 OS windows 7x64

Trace:
2017/02/14 05:27:29 [error] [php] Undefined variable: _COOKIE (C:\Program Files\Nectus\Web\Apache24\htdocs\yii\framework\web\CHttpRequest.php:1311)
Stack trace:
#0 C:\Program Files\Nectus\Web\Apache24\htdocs\yii\framework\web\auth\CWebUser.php(459): HttpRequest->getCookies()
#1 C:\Program Files\Nectus\Web\Apache24\htdocs\yii\framework\web\auth\CWebUser.php(201): CWebUser->restoreFromCookie()
#2 C:\Program Files\Nectus\Web\Apache24\htdocs\yii\framework\base\CModule.php(387): CWebUser->init()
#3 C:\Program Files\Nectus\Web\Apache24\htdocs\yii\framework\base\CModule.php(118): CWebApplication->getComponent()
#4 C:\Program Files\Nectus\Web\Apache24\htdocs\protected\controllers\DevicesBySitesController.php(48): CWebApplication->__isset()
#5 C:\Program Files\Nectus\Web\Apache24\htdocs\yii\framework\web\CWebApplication.php(359): DevicesBySitesController->__construct()
#6 C:\Program Files\Nectus\Web\Apache24\htdocs\yii\framework\web\CWebApplication.php(276): CWebApplication->createController()
#7 C:\Program Files\Nectus\Web\Apache24\htdocs\yii\framework\web\CWebApplication.php(141): CWebApplication->runController()
#8 C:\Program Files\Nectus\Web\Apache24\htdocs\yii\framework\base\CApplication.php(180): CWebApplication->processRequest()
#9 C:\Program Files\Nectus\Web\Apache24\htdocs\index.php(17): CWebApplication->run()
REQUEST_URI=/devicesBySites/get
in C:\Program Files\Nectus\Web\Apache24\htdocs\protected\controllers\DevicesBySitesController.php (48)
in C:\Program Files\Nectus\Web\Apache24\htdocs\index.php (17)

Could anybody help me to understand when and why it happens
Thanks

**Comment by samdark** [comment]
Since we have no idea why it may happen we can't, we can not fix it.

**Comment by rottriges** [comment]
Hello,

we have the same issue as mentioned above with apache 2.4 php 7.1 on a win7 64bit virtual machine. (see screenshots bellow)

![fehleryii](https://user-images.githubusercontent.com/12731068/30699245-79871e1a-9ee4-11e7-9bbc-17575b0a10ab.PNG)

![screenshot-2017-9-19 yii base errorexception](https://user-images.githubusercontent.com/12731068/30699273-8877492c-9ee4-11e7-96fa-68417db9f354.png)

In further time the whole stuff was running on a debian-linux apache server without any trouble. But due to a change of the ERP-Database from MySql to MSSQL we moved the system on a virtual win7 machine. First the system was running for a view month and than the crashes happens regulary. After reinstalling the apache server on the windows machine the system was running again for a view months but than the periodic crashes happens again.  We tried to change the settings in the /config/web.php to:
```
$config = [
    ...
    'components' => [
        'request' => [
            'enableCsrfValidation'=>false,
            'enableCookieValidation'=>false,
        ],
        ...
```

But without access. 
Regards
